creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -498,16 +499,19 @@ class SnapbackSM {
     // TODO change to chunked parallel
     await Promise.all(userReplicaSets.map(async (user) => {
       try {
-        const { wallet, endpoint: secondary } = user
+        const { wallet, primary, secondary1, secondary2, endpoint: secondary } = user
 
-        // TODO - throw on null wallet (is this needed?)
+        // Short-circuit if primary is not self - this function is meant to be called from primary to secondaries only
+        if (primary !== this.endpoint) {
```

Comment:
I actually don't think this check is necessary.  `userReplicaSets` is the `potentialSyncRequests` array, and effectively we only enqueue potential syncs to secondaries:
```
 if (primary === this.endpoint) {
          // filter out false-y values to account for incomplete replica sets
          const secondaries = ([secondary1, secondary2]).filter(Boolean)

          for (const secondary of secondaries) {
              // .. handling unhealthy node logic
              potentialSyncRequests.push({ ...nodeUser, endpoint: secondary })
          }

            // ... 
        }
```

Reply:
yeah agreed, this shouldn't really be hit - but it would lead to a corrupted state since the clock comparison checks assume it is primary, and the log msg is helpful. we'll know if anything is going wrong with that log msg

Reply:
i think it's always good to have safeguards and early exist as a sanity check. doesn't seem like a bad idea to have this in here